io.github.maddecoder.Classic-RBDOOM-3-BFG.yml

```diff
@@ -0,0 +1,139 @@
+app-id: io.github.maddecoder.Classic-RBDOOM-3-BFG
+runtime: org.freedesktop.Platform
+sdk: org.freedesktop.Sdk
+runtime-version: "22.08"
+command: DOOMBFA.sh
+sdk-extensions:
+    - org.freedesktop.Sdk.Extension.mono6
+build-options:
+    append-path: /usr/lib/sdk/mono6/bin
+    append-ld-library-path: /usr/lib/sdk/mono6/lib
+    append-pkg-config-path: /usr/lib/sdk/mono6/lib/pkgconfig
+
+finish-args:
+- --device=all
+- --socket=wayland
+- --socket=x11
+- --share=ipc
+- --share=network
+- --socket=pulseaudio
+- --filesystem=~/.var/app:ro
+- --filesystem=home
+- --filesystem=/run/media:ro
+- --filesystem=/media:ro
+
+cleanup:
+- /include
+- /lib/*.a
+- /lib/*.la
+- /lib/pkgconfig
+- /lib/cmake
+- /lib/debug
+- /share/man
+- /share/doc
+
+modules:
+- libgdi.yml
+- name: DoomBFA
+  buildsystem: cmake
+  subdir: neo
+  builddir: true
+  no-make-install: true
+  config-opts:
+   - --preset=linux-retail
+   - -B .
+   - -DMAKE_DLL=OFF
+   - -DPACKAGED=ON
+   - -DUSE_VCPKG=OFF
+   - -DALL_STATIC=OFF
+   - -DFFMPEG_FOUND=true
+   - -DFFMPEG_AVCODEC_INCLUDE_DIR=/app/include
+   - -DFFMPEG_LIBAVCODEC=/app/lib/libavcodec.so
+   - -DFFMPEG_LIBAVFORMAT=/app/lib/libavformat.so
+   - -DFFMPEG_LIBAVUTIL=/app/lib/libavutil.so
+   - -DFFMPEG_LIBSWRESAMPLE=/app/lib/libswresample.so
+   - -DFFMPEG_LIBSWSCALE=/app/lib/libswscale.so
+   - -DOPENAL_FOUND=true
+   - -DOPENAL_INCLUDE_DIR=/app/include
+   - -DOPENAL_LIBRARY=/app/lib/libopenal.so
+   - -DOPENAL_VERSION_STRING=1.23.1
+   - -DUSE_SYSTEM_ZLIB=false
+   - -DZLIB_FOUND=true
+   - -DZLIB_INCLUDE_DIRS=/app/include
+   - -DZLIB_LIBRARIES=/app/lib/libz.so
+   - -DUSE_SYSTEM_LIBPNG=false
+   - -DPNG_FOUND=true
+   - -DPNG_INCLUDE_DIRS=/app/include
+   - -DPNG_LIBRARIES=/app/lib/libpng.so
+   - -DUSE_SYSTEM_LIBJPEG=false
+   - -DJPEG_FOUND=true
+   - -DJPEG_INCLUDE_DIRS=/app/include
+   - -DJPEG_LIBRARIES=/app/lib64/libjpeg.so
+   - -DSDL2_INCLUDE_DIR=/app/include/SDL2
+   - -DSDL2_LIBRARY=/app/lib/libSDL2.so;/app/lib/libSDL2main.a
+  sources:
+  - type: git
+    url: https://github.com/MadDeCoDeR/Classic-RBDOOM-3-BFG.git
+    tag: 1.3.0-BFA-PATCHED
+    commit: 9044c7b78a015325b7ead958b026fbad66643458
+    disable-submodules: true
+  modules:
+    - shared-modules/glew/glew.json
+    - shared-modules/glu/glu-9.json
+    - shared-modules/SDL2/SDL2-with-libdecor.json
+    - openal.json
+    - ffmpeg.json
+    - zlib.json
+    - libpng.json
+    - libjpeg-turbo.json
+    - rapidjson.json
```

Comment:
separate manifest for dependencies, unless shared-modules or autogenerate is a nightmare in maintenance. Please avoid it.

Reply:
I really meant this.

Reply:
The best I can do is to try to use the build in rapidjson.
The other 2 are required to be obtained (System Libs must be avoided like the plague)

Reply:
this is not about building them. it's about the separated manifest files for them. It's unmaintainable.

Reply:
Are you sure?
Putting them in the main manifest might be worse (You are getting a manifest with more lines and can be more confusing)

Reply:
do you maintain 50+ flatpak?

Reply:
50+? From where that come from?

Reply:
I do. this is my point. But chose to ignore experience, you are doomed to repeat the mistakes.

Reply:
Ok, it's done.
Both libgdi, ffmpeg and OpenAL-Soft are now in the main manifest instead of being separate files